### DISCUSSION TITLE
Add support for Obsidian callouts

### DISCUSSION THREAD

**Discussion by jcdiv47** [epwalsh/obsidian.nvim#619] [Ideas]
First of all, thank you so much for the amazing and helpful plugin, now I am able to edit my Obsidian notes in Neovim if I want.

I was wondering if you would consider adding (at least a certain degree of) support for [Obsidian callouts](https://help.obsidian.md/Editing+and+formatting/Callouts), which I don't think the plugin does right now unless I miss something. 

```
> [!info]
> Here's a callout block.
> It supports **Markdown**, [[Internal link|Wikilinks]], and [[Embed files|embeds]]!
```

Basically what I would like is similar experience as editing a callout in Obsidian:

1. When I type something like `> [!info]` the line turns into an icon, in similar fashion to say `- [x]` in checkboxes.

![screenshot-20240607-171501](https://github.com/epwalsh/obsidian.nvim/assets/45190310/2ce24cfa-3260-454b-b888-2dba39510656)

2. When I hit enter/return to go to a new line inside a callout block, it would automatically add `> ` to the beginning of this new line. I presume a simple solution/workaround is some custom comment format options like 

```lua
vim.api.nvim_create_autocmd("FileType", {
	pattern = "markdown",
	callback = function()
		vim.opt_local.formatoptions:append("r") -- `<CR>` in insert mode
		vim.opt_local.formatoptions:append("o") -- `o` in normal mode
		vim.opt_local.comments = {
			"b:>", 
		}
	end,
})
```
(code snippet copied from [reddit](https://www.reddit.com/r/neovim/comments/1bis4h3/comment/kvmfjka/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button))

The plugin is super useful as is right now but I would really like the support for callouts if that is possible and not too much of a hassle. Please kindly let me know if you would consider it. Thanks again!

**Comment by epwalsh** [comment]
Hey @jcdiv47, I'm not sure when I'd be able to tackle this, but someone else might. Could you turn this into a feature request issue for greater visibility?

**Reply by jcdiv47** [reply]
Sure, I'll submit a feature request.
